api/client/ffs.go

```diff
@@ -21,6 +19,34 @@ type JobEvent struct {
 	Err error
 }
 
+// PushConfigOption mutates a push configuration.
+type PushConfigOption func(o *PushConfig)
+
+// PushConfig contains options for pushing a Cid configuration.
+type PushConfig struct {
+	Config            ff.CidConfig
+	HasConfig         bool
+	OverrideConfig    bool
+	HasOverrideConfig bool
+}
+
+// WithCidConfig overrides the Api default Cid configuration.
+func WithCidConfig(c ff.CidConfig) PushConfigOption {
+	return func(o *PushConfig) {
+		o.Config = c
+		o.HasConfig = true
+	}
+}
+
+// WithOverride allows a new push configuration to override an existing one.
+// It's used as an extra security measure to avoid unwanted configuration changes.
+func WithOverride(override bool) PushConfigOption {
+	return func(o *PushConfig) {
+		o.OverrideConfig = override
+		o.HasOverrideConfig = true
+	}
+}
+
```

Comment:
Here's the client side `PushConfig` and options that allow us to convey the information necessary to optionally set any option and encode that over gRPC.